app/Http/Controllers/Auth/RegisteredUserController.php

```diff
@@ -27,24 +31,50 @@ public function create(): View
      *
      * @throws \Illuminate\Validation\ValidationException
      */
-    public function store(Request $request): RedirectResponse
+    public function store(Request $request, ImageService $imageService): RedirectResponse
     {
+
         $request->validate([
             'name' => ['required', 'string', 'max:255'],
             'email' => ['required', 'string', 'lowercase', 'email', 'max:255', 'unique:'.User::class],
             'password' => ['required', 'confirmed', Rules\Password::defaults()],
-            'profile_image_url' => ['string','url', 'max:255', 'nullable']
+            'profile_image' => ['nullable', 'image', 'mimes:jpeg,png,jpg,gif', 'max:2048'],
         ]);
 
         $user = User::create([
             'name' => $request->name,
             'email' => $request->email,
             'password' => Hash::make($request->password),
-            'profile_image_url' => $request->profile_image_url ?? 'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png'
         ]);
 
         event(new Registered($user));
 
+        if($request->profile_image && $request->profile_image->extension()){
+
+            $success = false;
+
+            $extension = $request->profile_image->extension();
+            $filename = "profile-image" . '.' . $extension;
+            $folder = "user-{$user->id}";
+            $path = $folder . '/' . $filename;
+            $fileContents = $request->profile_image->get();
+            $path = $request->file('profile_image')->storeAs($folder, $filename, 'public');
+
+            try{
+                $success = Storage::disk('public')->put($path, $fileContents);
+            } catch(\Throwable $e) {
```

Comment:
refactor: the try catch block should encompass the whole of the method. combine lines 64 and 69, and invert the condition so you can get rid of the else statement
example:
if (!store-the-image) {
return back with errors
}